persistent/tests/test_persistence.py

```diff
@@ -1340,6 +1340,31 @@ def __init__(self):
         self.assertEqual(list(jar._loaded), [])
         self.assertEqual(list(jar._registered), [])
 
+    def test__p_invalidate_from_changed_w_slots_compat(self):
+        # check that (for backward-compatibility reason) slots are not released
+        # for classes where __new__ is overwritten.
+        class Derived(self._getTargetClass()):
+            __slots__ = ('myattr1', 'myattr2')
```

Comment:
It might be a good idea to add `__dict__` and set an attribute in it that's *not* in `__slots__` and then confirm that `__dict__` still gets cleared even when the slots aren't. (That's the circumstance that the contained proxy is in.)